app/src/main/java/org/oppia/app/databinding/TextViewBindingAdapters.java

```diff
@@ -0,0 +1,98 @@
+package org.oppia.app.databinding;
+
+import android.content.Context;
+import android.content.res.Resources;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.databinding.BindingAdapter;
+
+import org.oppia.app.R;
+import org.oppia.util.system.OppiaDateTimeFormatter;
+
+import java.util.Locale;
+
+/**
+ * Holds all the custom binding adapters that bind to [TextView] adapters.
+ */
+public final class TextViewBindingAdapters {
+  /**
+   * Binds date text with relative time.
+   */
+  @BindingAdapter("profile:created")
+  public static void setProfileDataText(@NonNull TextView textView, long timestamp) {
+    OppiaDateTimeFormatter oppiaDateTimeFormatter = new OppiaDateTimeFormatter();
+    String time = oppiaDateTimeFormatter.formatDateFromDateString(
+        OppiaDateTimeFormatter.DD_MMM_YYYY,
+        timestamp,
+        Locale.getDefault()
+    );
+    textView.setText(textView.getContext().getString(
+        R.string.profile_edit_created,
+        time
+    ));
+  }
+
+  @BindingAdapter("profile:lastVisited")
+  public static void setProfileLastVisitedText(@NonNull TextView textView, long timestamp) {
+    textView.setText(
+        String.format(
+            textView.getContext().getString(R.string.profile_last_used) + " " + getTimeAgo(
+                timestamp,
+                textView.getContext()
+            )
+        )
+    );
+  }
+
+  private static int SECOND_MILLIS = 1000;
+  private static int MINUTE_MILLIS = 60 * SECOND_MILLIS;
+  private static int HOUR_MILLIS = 60 * MINUTE_MILLIS;
+  private static int DAY_MILLIS = 24 * HOUR_MILLIS;
+
+  public static String getTimeAgo(long lastVisitedTimeStamp, Context context) {
+
+    OppiaDateTimeFormatter oppiaDateTimeFormatter = new OppiaDateTimeFormatter();
+    long timeStamp =
+        oppiaDateTimeFormatter.checkAndConvertTimestampToMilliseconds(lastVisitedTimeStamp);
+    long now = oppiaDateTimeFormatter.currentDate().getTime();
+
+    if (timeStamp > now || timeStamp <= 0) {
+      return "";
+    }
+
+    Resources res = context.getResources();
+    long timeDifference = now - timeStamp;
```

Comment:
Everywhere in this method: where there's a time variable, please make sure it's qualified with its unit (e.g. ``currentTimeMillis`` instead of ``now``). Mixing up time units can result in subtle bugs, so it's better to be specific.

Reply:
Done!